src/microbe_stage/editor/CellEditorComponent.cs

```diff
@@ -1902,6 +1909,39 @@ private void StartAutoEvoPrediction()
 
         UpdateCompoundLastingTimes(compoundBalanceData, nightBalanceData, nominalStorage,
             specificStorages ?? throw new Exception("Special storages should have been calculated"));
+
+        // Handle process list
+        var processes = new List<TweakedProcess>();
+        var processStatistics = new List<ProcessSpeedInformation>();
+
+        ProcessSystem.ComputeActiveProcessList(editedMicrobeOrganelles, ref processes);
+
+        float consumptionProductionRatio = energyBalance.TotalConsumption / energyBalance.TotalProduction;
+
+        foreach (var process in processes!)
+        {
+            var singleProcess = ProcessSystem.CalculateProcessMaximumSpeed(process, biome, CompoundAmountType.Current);
+
+            if (consumptionProductionRatio < 1.0)
+            {
+                foreach (var input in singleProcess.Inputs)
+                {
+                    if (input.Key == atp)
+                        continue;
+
+                    singleProcess.WritableInputs[input.Key] = input.Value * consumptionProductionRatio;
```

Comment:
Does this actually work? Doesn't this write to the same dictionary this is looping over?

Reply:
While WritableInputs write later in the code into Inputs, this somehow works.

Reply:
Well I'd like to find out why this works because if this suddenly stops working when someone else makes a completely unrelated change, that's going to be super unfun to that person making the other change.

Reply:
https://github.com/Revolutionary-Games/Thrive/blob/master/src/microbe_stage/ProcessSpeedInformation.cs#L29

Reply:
That creates an enumerator that is not immediately read. So that should run into exactly the same bug if a foreach loop was used as it also fundamentally also uses an enumerator on the dictionary.

Reply:
This was not addressed yet.

Reply:
I think this works only because all processes have at most 1 non-environmental input and output. I'll redo the code with a temp dictionary to avoid the problem.